frontend/src/pages/SettingsPage.js

```diff
@@ -1,19 +1,57 @@
-import React from "react";
-import Button from "@material-ui/core/Button";
-import TypeWarningSetting from "../components/settings/TypeWarningSetting";
+import React, {useState} from "react";
+// import { getSettings } from '../services/settingsApi'
+import SettingsForm from "../components/settings/SettingsForm";
 
 export default function SettingsPage() {
+    const [settings, setSettings] = useState({
+        warningCategory: {
+            warningA: true,
+            warningD: true,
+        },
+        distance: 15,
+        route: {
+            start: "",
+            end: "",
+            points: []
+        }
+    })
+
+    const [isLoading, setIsLoading] = useState(false);
+    const [error, setError] = useState(false);
+
+    const fetchData = async () => {
+        setIsLoading(true);
+        setError(null);
+
+        try {
+            // const resp = await getSettings();
+            // setSettings(resp);
+            setIsLoading(false);
+        } catch (e) {
+            setError(e);
+            setIsLoading(false);
+        }
+    };
+
     return (
         <section>
             <h2>Settings</h2>
             <p>
                 Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse malesuada lacus ex,
                 sit amet blandit leo lobortis eget.
             </p>
-            <form>
-                <TypeWarningSetting />
-                <Button variant="contained" color="primary" type="submit" >Save</Button>
-            </form>
+
+            {isLoading && <p>Loading...</p>}
+
+            {!isLoading && error ? (
+                <p>Oh no something went wrong!</p>
+
+            ) : null}
```

Comment:
```suggestion
            {!isLoading && error && <p>Oh no something went wrong!</p>}
```

No need for a ternary here

Reply:
ok